client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -73,7 +74,24 @@ def class_module_names(model_class: Type[VertaModelBase]) -> Set[str]:
     return modules_found
 
 
-# TODO VRD-682 convert module names to pkg names here
 def package_names(module_names: Set[str]) -> Set[str]:
-    """Return a set of all packages detected in the provided set of base module names."""
-    return module_names
+    """Return a set of distribution package names for all 3rd-party import
+    modules in the provided set.
+
+    Package distributions don't exist for modules in the Python standard library.
+    This fact is used to capture only 3rd-party packages.  Modules associated
+    with a single package name are easily captured.  Modules associated with a
+    namespace package (e.g.`google`), may come from a variety of distribution
+    packages.  To avoid making an arbitrary choice in those cases, the main
+    namespace package name is fetched from the distribution metadata.
+    """
+    pkg_dist = packages_distributions()
+    pkgs = set()
+    for mod in module_names:
+        dist_pkgs: Union[List[str], None] = pkg_dist.get(mod)
+        if dist_pkgs:
+            if len(dist_pkgs) == 1:
+                pkgs.add(dist_pkgs[0])
+            if len(dist_pkgs) > 1:
+                pkgs.add(distribution(mod).name)
```

Comment:
A "namespace distribution package" will have multiple options for packages that _could_ install the given module.

```python
from importlib.metadata import packages_distributions
pkgs = packages_distributions()
print(pkgs.get('google'))
```
Output:
`['googleapis-common-protos', 'google-cloud-bigquery', 'protobuf', 'google-cloud-core', 'google-api-core', 'google-resumable-media', 'google-auth']`
Hence why the length is checked and the two flavors are handled differently

Reply:
It doesn't seem that this works 🤔 

```python
distribution("google").name
```

raises

```
PackageNotFoundError: No package metadata was found for google
```

Reply:
I'm actually not sure what is meant to happen here. I think the logic we might want is "`pkgs.get('google')` returns a handful of potential packages for this module; the user's environment` should supply at least one of them to hopefully provide that module."

Reply:
`google` is a [namespace package](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/) , so you could run `pip install google` and get all the distribution packages under that umbrella.  In that case `packages_distributions().get('google')` returns the list of them all.  The challenge here is handling both cases:  User env has a single package installed that contains the required module OR user env has the whole namespace package.  I _think_ I've found a way to do that "gracefully" but it required tweaking return types and using dictionaries to track modules with their list of possible packages.  Check out the latest commits.

Reply:
Closing, since there are multiple threads on this issue